### ISSUE TITLE
can 2 job share the memory of 1 GPU?

### ISSUE THREAD

**Issue by Jong-Won** [microsoft/pai#2946]
Can PAI support that 2 job sharing the memory 1 GPU? I have not find the function in the document.

Here is a reference https://yq.aliyun.com/articles/690623

**Comment by abuccts** [comment]
No, and it is not recommended.

It is possible in theory of course, but in practice it won't provide any benefit.
Here're the two possible approaches:
* Time sharing. Swapping in GPU costs time and it will be slower than running in a queue.
* Divide GPU cores and memory. It's difficult to isolate memory in one card and CUDA does not support it. Fewer memory will also slow down the speed of training or inference.

**Comment by Jong-Won** [comment]
@abuccts 

Thank you for your reply.
Lets suppose these is only one card and one deep learning program that occupy less than 1/2 GPU memory and GPU ulitize is also less than 1/2. Our task is execute the program twice.

Do you mean that execute deep learning program one by one (pipline) is more efficient in time than the way we execute them together simultaneously in same GPU card in practice ?

Thank you!

**Comment by abuccts** [comment]
It depends on your workload.

In most cases, if your deep learning job occupies less than 1/2 GPU memory and utilization, the best solution is to optimize your job to make the full use of whole GPU, e.g. double the batch size for training / inference, improve IO performance to maximum GPU utilization.

For example, your task is inference twice using a DNN model, you can simply double the batch size and merge two inference jobs into one, which will be more efficient than executing two jobs simultaneously.
Maybe your workload is special, you can also share it here for discussion.

**Comment by Jong-Won** [comment]
I think increasing batch size is good way during inference. However, batch size is a super-parameter, which can effect the accurancy. In this situation, we cannot change the batch size. The process may occupy the whole GPU even if it actually cost less than half memory and utilization of GPU. Am I right?

**Comment by abuccts** [comment]
Yes, the GPU card is allocated entirely.
For training, you can try `0.5 * learning_rate` when you use `2 * batch_size` to keep accuracy.

**Comment by abuccts** [comment]
Yes, the GPU card is allocated entirely.
For training, you can try `k * learning_rate` and `1/k * steps_number` when you use `k * batch_size` to keep accuracy.
